File: semester7/EYzEES/Lab3/abstract-builder/src/main/java/com/software/bsuir/abstractbuilder/clients/HuggingfaceClient.java
package com.software.bsuir.abstractbuilder.clients;

import com.software.bsuir.abstractbuilder.entities.GeneralAbstract;
import com.software.bsuir.abstractbuilder.entities.HuggingFaceRequestBody;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;

import java.util.List;
import java.util.Map;

@FeignClient(name = "huggingface")
public interface HuggingfaceClient {
    // /facebook/bart-large-cnn
    @PostMapping("/sshleifer/distilbart-cnn-12-6")
    ResponseEntity<List<GeneralAbstract>> buildAbstractsWithModel(
            @RequestHeader("Authorization") String token,
            @RequestBody HuggingFaceRequestBody huggingFaceRequestBody
    );
}

File: semester7/EYzEES/Lab2/language-guide/src/main/kotlin/org/example/languageguide/clients/HuggingfaceClient.kt
package org.example.languageguide.clients

import org.example.languageguide.config.FeignConfig
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestHeader


@FeignClient(name = "hugging-face", url = "https://api-inference.huggingface.co")
interface HuggingfaceClient {
    @PostMapping("/models/papluca/xlm-roberta-base-language-detection")
    fun detectLanguage(@RequestBody requestBody: Map<String, Any>,
                       @RequestHeader("Authorization") authorization: String): String
}